### PR TITLE
feat: Add examples to the spec JSON schema

### PR DESCRIPTION
**Pull request by candiduslynx**
Part of https://github.com/cloudquery/cloudquery-issues/issues/1401

### CODE REVIEW

File: plugins/source/azure/client/spec/spec.go [hunk 1]
```diff
@@ -29,19 +29,19 @@ type Spec struct {
 	// An OIDC token can be used to authenticate with Azure instead of `AZURE_CLIENT_SECRET`.
 	// This is useful for Azure AD workload identity federation.
 	// When using this option, the `AZURE_CLIENT_ID` and `AZURE_TENANT_ID` environment variables must be set.
-	OIDCToken string `json:"oidc_token" jsonschema:"minLength=1"`
+	OIDCToken string `json:"oidc_token" jsonschema:"minLength=1,example=oidc_token"`
 
 	// The best effort maximum number of Go routines to use.
 	// Lower this number to reduce memory usage.
-	Concurrency int `json:"concurrency" jsonschema:"minimum=1,default=50000"`
+	Concurrency int `json:"concurrency" jsonschema:"minimum=1,default=50000,example=50000"`
```
Comment (erezrokah) [inline plugins/source/azure/client/spec/spec.go:36]:
```suggestion
	Concurrency int `json:"concurrency" jsonschema:"minimum=1,default=50000"`
```

File: plugins/source/azure/client/spec/spec.go [hunk 2]
```diff
@@ -29,19 +29,19 @@ type Spec struct {
...
 
 	// During initialization the Azure source plugin discovers all resource groups
 	// and enabled resource providers per subscription, to be used later on during the sync process.
 	// The plugin runs the discovery process in parallel.
 	// This setting controls the maximum number of concurrent requests to the Azure API during discovery.
 	// Only accounts with many subscriptions should require modifying this setting,
 	// to either lower it to avoid network errors, or to increase it to speed up the discovery process.
-	DiscoveryConcurrency int `json:"discovery_concurrency" jsonschema:"minimum=1,default=400"`
+	DiscoveryConcurrency int `json:"discovery_concurrency" jsonschema:"minimum=1,default=400,example=400"`
```
Comment (erezrokah) [inline plugins/source/azure/client/spec/spec.go:44]:
```suggestion
	DiscoveryConcurrency int `json:"discovery_concurrency" jsonschema:"minimum=1,default=400"`
```

File: plugins/source/azure/client/spec/spec.go (hunk 1 above)
Comment (candiduslynx) [reply plugins/source/azure/client/spec/spec.go]: b26919b47456c26b8292d9b08a6ed444e1952217

File: plugins/source/azure/client/spec/spec.go (hunk 2 above)
Comment (candiduslynx) [reply plugins/source/azure/client/spec/spec.go]: b26919b47456c26b8292d9b08a6ed444e1952217

File: plugins/source/azure/client/spec/spec.go [hunk 3]
```diff
@@ -12,15 +12,15 @@ import (
 type Spec struct {
 	// Specify which subscriptions to sync data from.
 	// Empty means all visible subscriptions.
-	Subscriptions []string `json:"subscriptions" jsonschema:"minLength=1,uniqueItems=true"`
+	Subscriptions []string `json:"subscriptions" jsonschema:"minLength=1,uniqueItems=true,example=subscription_id"`
```
Comment (hermanschaaf) [inline plugins/source/azure/client/spec/spec.go:15]: Shouldn't the examples be real examples, like `00000000-0000-0000-0000-000000000000`?

File: plugins/source/azure/client/spec/spec.go [hunk 4]
```diff
@@ -12,15 +12,15 @@ import (
...
 
 	// A list of subscription IDs that CloudQuery will skip syncing.
 	// This is useful if CloudQuery is discovering the list of subscription IDs and there are some subscriptions that you want to not even attempt syncing.
-	SkipSubscriptions []string `json:"skip_subscriptions" jsonschema:"minLength=1,uniqueItems=true"`
+	SkipSubscriptions []string `json:"skip_subscriptions" jsonschema:"minLength=1,uniqueItems=true,example=subscription_id"`
 
 	// The name of the cloud environment to use.
 	// See the [Azure CLI documentation](https://learn.microsoft.com/en-us/cli/azure/manage-clouds-azure-cli) for more information.
-	CloudName string `json:"cloud_name" jsonschema:"minLength=1"`
+	CloudName string `json:"cloud_name" jsonschema:"minLength=1,example=cloud_name"`
```
Comment (erezrokah) [inline plugins/source/azure/client/spec/spec.go:23]:
```suggestion
	CloudName string `json:"cloud_name" jsonschema:"minLength=1,example=AzureCloud"`
```

File: plugins/source/azure/client/spec/spec.go (hunk 3 above)
Comment (candiduslynx) [inline plugins/source/azure/client/spec/spec.go:15]:
```suggestion
	Subscriptions []string `json:"subscriptions" jsonschema:"minLength=1,uniqueItems=true,example=00000000-0000-0000-0000-000000000000"`
```

File: plugins/source/azure/client/spec/spec.go [hunk 5]
```diff
@@ -12,15 +12,15 @@ import (
 type Spec struct {
 	// Specify which subscriptions to sync data from.
 	// Empty means all visible subscriptions.
-	Subscriptions []string `json:"subscriptions" jsonschema:"minLength=1,uniqueItems=true"`
+	Subscriptions []string `json:"subscriptions" jsonschema:"minLength=1,uniqueItems=true,example=subscription_id"`
 
 	// A list of subscription IDs that CloudQuery will skip syncing.
 	// This is useful if CloudQuery is discovering the list of subscription IDs and there are some subscriptions that you want to not even attempt syncing.
-	SkipSubscriptions []string `json:"skip_subscriptions" jsonschema:"minLength=1,uniqueItems=true"`
+	SkipSubscriptions []string `json:"skip_subscriptions" jsonschema:"minLength=1,uniqueItems=true,example=subscription_id"`
```
Comment (candiduslynx) [inline plugins/source/azure/client/spec/spec.go:19]:
```suggestion
	SkipSubscriptions []string `json:"skip_subscriptions" jsonschema:"minLength=1,uniqueItems=true,example=00000000-0000-0000-0000-000000000000"`
```

**Comment by candiduslynx** [comment]
/gen sha=eb891e4985ab06ad868befe627d4428e297808b2 dir=plugins/source/azure

**Review by erezrokah — APPROVED** [review]

**Pull request merged**